include/tvm/runtime/c_runtime_api.h

```diff
@@ -559,6 +559,20 @@ TVM_DLL int TVMByteArrayFree(TVMByteArray* arr);
 TVM_DLL int TVMDeviceAllocDataSpace(DLContext ctx, size_t nbytes, size_t alignment,
                                     DLDataType type_hint, void** out_data);
 
+/*!
+ * \brief Allocate a data space on device.
```

Comment:
```
Allocate a data space on device with special memory scope. 

\note The memory could use a special multi-dimensional memory layout. That is why we pass shape,
         and dtype instead of raw number of bytes.
```